Source/WebKit/WebProcess/Extensions/API/WebExtensionAPISidebarAction.h

```diff
@@ -35,6 +35,33 @@ OBJC_CLASS NSString;
 
 namespace WebKit {
 
+using SidebarError = NSString *;
+// In this variant, `monostate` indicates that we have neither a window or tab identifier, but no error
+using ParseResult = std::variant<std::monostate, WebExtensionTabIdentifier, WebExtensionWindowIdentifier, SidebarError>;
+
+template<typename T, typename VARIANT_T>
+struct isVariantMember;
+template<typename T, typename... ALL_T>
+struct isVariantMember<T, std::variant<ALL_T...>> : public std::disjunction<std::is_same<T, ALL_T>...> { };
```

Comment:
These generic helpers should go somewhere else, but I'm not sure where…

Reply:
I had the same dilemma. To me, they feel too general for the SidebarAction header, but not quite general enough for CocoaHelpers.